<?xml version="1.0" encoding="utf-8"?>
<launch>
  <arg name="node_name" default="objectness" />

  <arg name="image" default="image" doc="The image topic. Should be remapped to the name of the real image topic." />

  <arg name="use_camera_info" default="false" doc="Indicates that the camera_info topic should be subscribed to to get the default input_frame_id. Otherwise the frame from the image message will be used." />
  <arg name="debug_view" default="true" doc="Specify whether the node displays a window to show image" />
  <arg name="queue_size" default="3" doc="Specigy queue_size of input image subscribers" />

  <arg name="training_path" default="" doc="The path of trained model.
                                            Please download
                                            https://github.com/opencv/opencv_contrib/tree/3.4/modules/saliency/samples/ObjectnessTrainedModel"     />
  <arg name="nss" default="2" doc="Size for non-maximal suppress" />
  <arg name="max_objectness" default="10" doc="Maximum number of objectness to be detected" />

  <!-- equalize_histogram.cpp  -->
  <node name="$(arg node_name)" pkg="opencv_apps" type="objectness" output="screen">
    <remap from="image" to="$(arg image)" />
    <param name="use_camera_info" value="$(arg use_camera_info)" />
    <param name="debug_view" value="$(arg debug_view)" />
    <param name="queue_size" value="$(arg queue_size)" />
    <param name="training_path" value="$(arg training_path)" />
    <param name="nss" value="$(arg nss)" />
    <param name="max_objectness" value="$(arg max_objectness)" />
  </node>
</launch>
